Userland/Libraries/LibWeb/HTML/DocumentState.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/URL.h>
+#include <LibJS/Heap/Cell.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/Origin.h>
+#include <LibWeb/HTML/PolicyContainers.h>
+#include <LibWeb/ReferrerPolicy/ReferrerPolicy.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#document-state-2
+class DocumentState final : public JS::Cell {
+    JS_CELL(DocumentState, JS::Cell);
+
+public:
+    virtual ~DocumentState();
+
+    // https://html.spec.whatwg.org/multipage/browsing-the-web.html#document-state-document
+    JS::GCPtr<DOM::Document> document;
```

Comment:
Class with public data members that are not exposed via getters is very much against https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#names:

> Data members in C++ classes should be private. [...] data members should be prefixed by "m_".